libbeat/processors/actions/alterFieldProcessor.go

```diff
@@ -133,3 +149,22 @@ func (a *alterFieldProcessor) alter(event *beat.Event, field string) error {
 
 	return nil
 }
+
+func (a *alterFieldProcessor) alterValue(event *beat.Event, value string) error {
```

Comment:
Better to rename, so it's clear this not the actual value.

```suggestion
func (a *alterFieldProcessor) alterValue(event *beat.Event, valueKey string) error {
```

This does not look like the implementation we have discussed before.

We need to:

1. Get a value for `valueKey`
2. Check if it's a string
3. If it's not a string – throw error
4. If it's a string replace it with the altered value.

For this I don't see how it's necessary to use `Traverse`, why not just `GetValue,Delete,Put`? If the end value is always a string and we don't change the `valueKey` it should be way simpler.

Is there something I'm missing that it has to use `Traverse`?

Reply:
yes we can use `GetValue`, `Delete` and `Put`. Just both seemed okay

Reply:
I think when we don't need to modify fields and we match case-sensitively, we should just use `GetValue`, `Delete`, `Put`.